crates/runtime/src/accelerated_table/refresh_task.rs

```diff
@@ -187,55 +185,65 @@ impl RefreshTask {
         })
     }
 
-    async fn run_once(&self) -> Result<(), RetryError<super::Error>> {
-        self.mark_dataset_status(status::ComponentStatus::Refreshing)
-            .await;
-
-        let dataset_name = self.dataset_name.clone();
-
-        let mode = self.refresh.read().await.mode;
+    async fn run_once(&self, request: &Refresh) -> Result<(), RetryError<super::Error>> {
+        Self::mark_dataset_status(
+            &self.dataset_name,
+            request.sql.as_deref(),
+            status::ComponentStatus::Refreshing,
+        );
 
         let _timer = TimeMeasurement::new(
-            match mode {
+            match request.mode {
                 RefreshMode::Disabled => {
                     unreachable!("Refresh cannot be called when acceleration is disabled")
                 }
                 RefreshMode::Full => &metrics::LOAD_DURATION_MS,
                 RefreshMode::Append => &metrics::APPEND_DURATION_MS,
                 RefreshMode::Changes => unreachable!("changes are handled upstream"),
             },
-            vec![Key::from_static_str("dataset").string(dataset_name.to_string())],
+            vec![Key::from_static_str("dataset").string(self.dataset_name.to_string())],
         );
 
         let start_time = SystemTime::now();
 
-        let get_data_update_result = match mode {
+        let get_data_update_result = match request.mode {
             RefreshMode::Disabled => {
                 unreachable!("Refresh cannot be called when acceleration is disabled")
             }
-            RefreshMode::Full => self.get_full_update().await,
-            RefreshMode::Append => self.get_incremental_append_update().await,
+            RefreshMode::Full => {
+                self.get_full_or_incremental_append_update(request, None)
```

Comment:
Should we use the name `refresh` instead of `request` here to keep consistent with the definition? Same for the  `get_full_or_incremental_append_update` below as well ` pub async fn get_full_or_incremental_append_update(
        &self,
        refresh: &Refresh,
        overwrite_timestamp_in_nano: Option<u128>,
    )`